src/Model/PropertyGo.cs

```diff
@@ -32,15 +32,7 @@ public string Tag(bool omitEmpty = true)
             bool hasParent = false;
             if (Parent is CompositeTypeGo go && !go.IsWrapperType)
             {
-                // This extra check is necessary because of https://github.com/Azure/autorest/issues/2889
-                if (this.IsPolymorphicDiscriminator)
-                {
-                    sb.Append(SerializedName);
-                }
-                else
-                {
-                    sb.Append(XmlName);
-                }
+                sb.Append(XmlName);

```

Comment:
is this not producing issues anymore ? I think for storage files sdk it was showing up.

Reply:
I rebuilt blob, files and queues with this change and it worked as expected (i.e. no crash).

Reply:
ok if `go build` on them was fine also, I think its good